File: mobile/flutter/lib/crypto_wallet.dart
class BizXWallet {
  static const supportedCoins = ['BTC','BCH','LTC','DOGE','ETH','ETC','SOL','ADA','XRP','DOT','AVAX'];
  final String coin, address; final bool watchOnly;
  BizXWallet(this.coin, this.address, {this.watchOnly = true}) : assert(supportedCoins.contains(coin));
  Map<String,String> transaction(String to, String amount) => {'state':'awaiting_confirmation','coin':coin,'from':address,'to':to,'amount':amount};
}

File: apple/CryptoWallet.swift
import Foundation
public struct BizXWallet {
    public static let supportedCoins = ["BTC","BCH","LTC","DOGE","ETH","ETC","SOL","ADA","XRP","DOT","AVAX"]
    public let coin: String; public let address: String; public let watchOnly: Bool
    public init(coin: String, address: String, watchOnly: Bool = true) { precondition(Self.supportedCoins.contains(coin)); self.coin=coin; self.address=address; self.watchOnly=watchOnly }
    public func transaction(to: String, amount: String) -> [String:String] { ["state":"awaiting_confirmation","coin":coin,"from":address,"to":to,"amount":amount] }
}
